lib/spack/spack/util/module_cmd.py

```diff
@@ -31,7 +33,31 @@ def module(*args):
     if args[0] in module_change_commands:
         # Do the module manipulation, then output the environment in JSON
         # and read the JSON back in the parent process to update os.environ
-        module_cmd += ' >/dev/null; python -c %s' % py_cmd
+        # For python, we use the same python running the Spack process, because
+        # we can guarantee its existence. We have to do some LD_LIBRARY_PATH
+        # shenanigans to ensure python will run.
+
+        # LD_LIBRARY_PATH under which Spack ran
+        os.environ['SPACK_LD_LIBRARY_PATH'] = spack.main.spack_ld_library_path
```

Comment:
I think this still has a problem.

If we load multiple modules, we will only get the changes to LD_LIBRARY_PATH from the last one loaded.

Reply:
We need to somehow diff the LD_LIBRARY_PATHS and apply the diff, not overwrite.